### PR TITLE
test/ceph_crypto: define __has_feature if the compiler doesn't have it

### PR DESCRIPTION
**Pull request by xxhdx1985126**
Fixes: https://tracker.ceph.com/issues/65673
Signed-off-by: Xuehan Xu <xuxuehan@qianxin.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/test/ceph_crypto.cc [hunk 1]
```diff
@@ -269,6 +269,11 @@ void do_simple_crypto() {
 }
 
 #if GTEST_HAS_DEATH_TEST && !defined(_WIN32)
+
+#ifndef __has_feature
```
Comment (tchaikov) [inline src/test/ceph_crypto.cc:273]: @xxhdx1985126 according to https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005ffeature.html , 

> Note that __has_feature and __has_extension are not recommended for use in new code, and are only provided for compatibility with Clang

i'd sugget reference https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005ffeature.html and https://clang.llvm.org/docs/LanguageExtensions.html#has-feature-and-has-extension in the commit message.

Comment (xxhdx1985126) [reply src/test/ceph_crypto.cc:273]: Done, please take a look again:-)

**Review by tchaikov — APPROVED** [review]
lgtm. but GCC-13 should have it defined though.

**Comment by xxhdx1985126** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
```
Took [3 min 47 sec](https://jenkins.ceph.com/job/ceph-pull-requests/buildTimeTrend) on
   [172.21.3.114+mira013](https://jenkins.ceph.com/computer/172.21.3.114+mira013)

No space left on device
```

**Comment by xxhdx1985126** [comment]
jenkins test make check

**Comment by xxhdx1985126** [comment]
jenkins test make check

**Pull request merged**